typed-racket-doc/typed-racket/scribblings/reference/experimental.scrbl

```diff
@@ -134,3 +139,57 @@ detects that the comparison is guaranteed to produce
 @racket[#f], and thus the clearly ill-typed `then'-branch is
 ignored by the type checker since it is guaranteed to be
 dead code.
+
+
+@section{Dependent Function Types}
+
+Typed Racket also supports explicitly dependent function types:
+  
+@defform*/subs[#:link-target? #f #:id -> #:literals (:)
+               [(-> ([id : opt-deps type opt-where] ...) type opt-props)]
```

Comment:
I talked with Sam about this and  punted the decision making aspect to him =)

I personally prefer re-using `->` since the presence of `:` in the domain clauses allows us to unambiguously distinguish dependent and non-dependent functions (TR reserves `:`, so users cannot meaningfully use it except in a very limited number of places). YMMV.